src/group_theory/submonoid/operations.lean

```diff
@@ -386,6 +386,15 @@ end galois_insertion
 @[to_additive "An `add_submonoid` of an `add_monoid` inherits an addition."]
 instance has_mul : has_mul S := ⟨λ a b, ⟨a.1 * b.1, S.mul_mem a.2 b.2⟩⟩
 
+/-- A monoid is isomorphic to itself as a submonoid. -/
+@[to_additive "An additive monoid is isomorphic to itself as an additive submonoid.", simps]
+def top_equiv : (⊤ : submonoid M) ≃* M :=
```

Comment:
Can you add the `simp` lemma that `top_equiv.to_monoid_hom = (⊤ : submonoid M).subtype`? It should be true by `monoid_hom.ext $ λ _ _, rfl`